packages/tcore-sdk/src/Types/Common/Common.types.ts

```diff
@@ -52,7 +52,8 @@ export const zQuery = z.object({
   amount: z
     .preprocess(preprocessNumber, z.number())
     .nullish()
-    .transform((x) => x ?? 20),
+    .transform((x) => x ?? 20)
+    .transform((x) => Math.max(x, 0)),
```

Comment:
Also improved the back-end validation to prevent negative numbers being passed to database modules.

Reply:
I chose two `transform`s to separate the concerns, the first one is the falsy validation, the second one is a negative/positive validation.